datalad/metadata/__init__.py

```diff
@@ -193,7 +193,25 @@ def get_metadata(ds, guess_type=False, ignore_subdatasets=False,
     # XXX it may be worth the put the combined output of this function in a separate
     # cache on the local machine, in order to speed up meta data access, but maybe this
     # is already the domain of a `query` implementation
-    meta.append(get_implicit_metadata(ds, ds_identifier))
+    implicit_meta = get_implicit_metadata(ds, ds_identifier)
+    # create a lookup dict to find parts by subdataset mountpoint
+    has_part = implicit_meta.get('dcterms:hasPart', [])
+    if not isinstance(has_part, list):
+        has_part = [has_part]
+    has_part = {hp['location']: hp for hp in has_part}
+
+    # figure out all other version of this dataset: origin or siblings
+    # build a flat list of UUIDs
+    candidates = ('dcterms:isVersionOf', 'dcterms:hasVersion')
+    ds_versions = [
+        # flatten list
+        item for versionlist in
+        # extract uuids of all listed repos
+        [[implicit_meta[i]['@id'] for r in implicit_meta[i] if '@id' in r]
```

Comment:
Hmm, likely you are seeing this more clearly than I do. Just want to point out that this flattens a nested list. I am in an intense conversion with an amazing bottle of Gin Tonic. Will stop converting it into code for now ;-)